sensor/common/scan/scan.go

```diff
@@ -78,28 +91,39 @@ func NewLocalScan(registryStore *registry.Store) *LocalScan {
 //
 // If no registry credentials are found an empty registry slice is passed to enrichLocalImageFromRegistry for enriching with 'no auth'.
 func (s *LocalScan) EnrichLocalImageInNamespace(ctx context.Context, centralClient v1.ImageServiceClient, ci *storage.ContainerImage, namespace string, requestID string, force bool) (*storage.Image, error) {
-	var regs []registryTypes.Registry
-	var reg registryTypes.Registry
-	var err error
-
 	imgName := ci.GetName()
 
+	regs := s.getRegistries(namespace, imgName)
+
+	log.Debugf("Attempting image enrich for %q in namespace %q with %v regs", ci.GetName().GetFullName(), namespace, len(regs))
+
+	return s.enrichLocalImageFromRegistry(ctx, centralClient, ci, regs, requestID, force)
+}
+
+func (s *LocalScan) getRegistries(namespace string, imgName *storage.ImageName) []registryTypes.ImageRegistry {
+	var regs []registryTypes.ImageRegistry
+
+	// Add registries from Central's image integrations.
+	centralIntegrations := s.getMatchingCentralRegIntegrations(imgName)
+	if len(centralIntegrations) > 0 {
+		regs = append(regs, centralIntegrations...)
+	}
+
+	// Add registries from k8s pull secrets.
 	if namespace != "" {
 		// if namespace provided pull appropriate registry.
-		reg, err := s.getRegistryForImageInNamespace(imgName, namespace)
-		if err == nil {
+		if reg, err := s.getRegistryForImageInNamespace(imgName, namespace); err == nil {
 			regs = append(regs, reg)
 		}
 	}
 
-	reg, err = s.getGlobalRegistryForImage(imgName)
+	// Add global pull secret registry
+	reg, err := s.getGlobalRegistryForImage(imgName)
```

Comment:
also, why is the error not important

Reply:
The registry store methods return an error when no registry is found, otherwise return a registry. The lack of an error indicates that a registry was found.

As part of ROX-16821 this logic will be changed, I just added a note change the registry_store methods to use a bool instead of error to indicate 'exists'